File: 268-missing-number/268-missing-number.c
int missingNumber(int* nums, int numsSize){
    int k = 0;
    int *a;
    
    a = (int*)calloc(sizeof(int) , numsSize+1);  //all elements are 0
    
    while(k < numsSize)
        a[nums[k++]] = 1;
    
    k = 0;
    while(k < numsSize+1)
    {
        if(a[k] == 0)
            return k;
        k++;
    }
    return k+1;
}

File: 268-missing-number/268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int len = nums.length;
        int sumOfArray = (len * (len+1))/2;
        for(int i = 0; i < len; i++) 
            sumOfArray -= nums[i];
        return sumOfArray;
    }   
}
//Time: O(n) , Memory: O(1)